src/test/kotlin/com/vanniktech/rxriddles/solutions/Riddle6Solution.kt

```diff
@@ -2,8 +2,13 @@ package com.vanniktech.rxriddles.solutions
 
 import io.reactivex.Single
 import io.reactivex.functions.BiFunction
+import io.reactivex.schedulers.Schedulers
 
 object Riddle6Solution {
   fun solve(first: Single<Int>, second: Single<Int>)
-      = Single.zip(first, second, BiFunction<Int, Int, Pair<Int, Int>> { t1, t2 -> t1 to t2 })
+      = Single.zip(
```

Comment:
What if you apply `subscribeOn` to only one of them? Like:
```
Single.zip(
      first.subscribeOn(Schedulers.io()),
      second,
      BiFunction{...})
```
Is that parallel enough? Should it pass tests?

Reply:
Good question, and I don't know. I'm inclined to add a check that both observables are susbcribed to off-thread and adjust the comment for the riddle state that both observables should be triggered _in background_. Thoughts?